src/Integration.Vsix.UnitTests/Suppression/LiveIssueFactoryTests.cs

```diff
@@ -59,31 +60,60 @@ public void Ctor_WithNullVsSolution_ThrowsArgumentNullException()
         public void Ctor_WithNonNullParameters_ReadProjectGuidOfAllProjectsInSolution()
         {
             // Arrange
-            var vsSolutionMock = new Mock<IVsSolution>();
-            uint fileCount = 2; // initialized with a value to actually assign a value to the out
-            vsSolutionMock.Setup(x => x.GetProjectFilesInSolution(0, 0, null, out fileCount))
-                .Returns(VSConstants.S_OK);
-            var fileNames = new string[fileCount];
-            vsSolutionMock.Setup(x => x.GetProjectFilesInSolution(0, fileCount, fileNames, out fileCount))
-                .OutCallback((uint x, uint y, string[] names, out uint z) =>
-                    {
-                        z = 0;
-                        names[0] = "Project1";
-                        names[1] = "Project2";
-                    });
-            vsSolutionMock.As<IVsSolution5>().Setup(x => x.GetGuidOfProjectFile(It.IsAny<string>()))
-                .Returns(Guid.Empty);
+            uint fileCountOut;
+            var vsSolutionMock = SetupSolutionMocks(
+                new KeyValuePair<string, string>("Project1", "11111111-1111-1111-1111-111111111111"),
+                new KeyValuePair<string, string>("Project2", "22222222-2222-2222-2222-222222222222"));
 
             // Act
             new LiveIssueFactory(new AdhocWorkspace(), vsSolutionMock.Object);
 
             // Assert
-            vsSolutionMock.Verify(x => x.GetProjectFilesInSolution(0, It.IsAny<uint>(), It.IsAny<string[]>(), out fileCount),
+            vsSolutionMock.Verify(x => x.GetProjectFilesInSolution(0, It.IsAny<uint>(), It.IsAny<string[]>(), out fileCountOut),
                 Times.Exactly(2));
             vsSolutionMock.As<IVsSolution5>().Verify(x => x.GetGuidOfProjectFile(It.IsAny<string>()), Times.Exactly(2));
         }
 
         [TestMethod]
+        public void BuildMap_DuplicateSolutionFolderNamesAreIgnored()
+        {
+            // See #413: https://github.com/SonarSource/sonarlint-visualstudio/issues/413
+
+            // Arrange
+            var vsSolutionMock = SetupSolutionMocks(
+                new KeyValuePair<string, string>("SolutionFolder1", "11111111-1111-1111-1111-111111111111"),
+                new KeyValuePair<string, string>("Item1", "22222222-2222-2222-2222-222222222222"),
+                new KeyValuePair<string, string>("SolutionFolder2", "33333333-3333-3333-3333-333333333333"),
+                new KeyValuePair<string, string>("Item1", "44444444-4444-4444-4444-444444444444"),
+                new KeyValuePair<string, string>("realProject.csproj", "55555555-5555-5555-5555-555555555555"),
```

Comment:
Real project files will return the fully-qualified path to the file, so there can't be any duplicates. However, solution folders are also treated as projects by VS, and the "path" that is returned is just the simple name. This means it's possible to have duplicates if there are nested folders e.g. _Folder1\ItemX_, _Folder2\ItemX_ - both will return a "path" of _ItemX_.